checkov/common/runners/object_runner.py

```diff
@@ -104,13 +118,20 @@ def run(
                     code_block=definitions_raw[file_path][start - 1:end + 1],
                     file_path=f"/{os.path.relpath(file_path, root_folder)}",
                     file_line_range=[start, end + 1],
-                    resource=self.get_resource(file_path, key, check.supported_entities),  # type:ignore[arg-type]  # key is str not BaseCheck
+                    resource=self.get_resource(file_path, key, check.supported_entities),# type:ignore[arg-type]  # key is str not BaseCheck
                     evaluations=None,
                     check_class=check.__class__.__module__,
                     file_abs_path=os.path.abspath(file_path),
                     entity_tags=None,
                     severity=check.severity,
                 )
+                if self.check_type == CheckType.GITHUB_ACTIONS:
+                    record = GithubActionsRecord(
+                        record=record,
```

Comment:
@NoaAzoulay IMO, it is better and cleaner to not use the class you are extending inside your own class constructor.
If we had mypy on this directory I would feel more confident, but we don't. So my suggestion - check what's your check_type, and if it is GHA create a GHARecord with all the fields **set one by one**. If not - create the regular record.
The constructor doing the spreading for its super is... harder to maintain, and we've broken it a few times already